### PR TITLE
[OPENJDK-2411] JDK21 images for UBI8

### PR DESCRIPTION
**Pull request by jmtd**
https://issues.redhat.com/browse/OPENJDK-2411

### CODE REVIEW

**Comment by jmtd** [comment]
I am doing a round of scratch builds for these and running the behave test suite locally.

**Comment by jmtd** [comment]
> I am doing a round of scratch builds for these and running the behave test suite locally.

all passed

File: ubi8-openjdk-21.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+- name: "io.k8s.display-name"
+  value: "Java Applications"
+- name: "io.openshift.tags"
+  value: "builder,java"
+- name: "maintainer"
+  value: "Red Hat OpenJDK <openjdk@redhat.com>"
+- name: "com.redhat.component"
+  value: "openjdk-21-ubi8-container"
+- name: "usage"
+  value: "https://access.redhat.com/documentation/en-us/red_hat_jboss_middleware_for_openshift/3/html/red_hat_java_s2i_for_openshift/"
```
Comment (jerboaa) [inline ubi8-openjdk-21.yaml]: Should this point to?

https://jboss-container-images.github.io/openjdk/

File: ubi8-openjdk-21.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+modules:
+  repositories:
+  - path: modules
+  install:
+  - name: jboss.container.util.pkg-update
+  - name: jboss.container.openjdk.jdk
+    version: "21"
+  - name: jboss.container.maven
+    version: "3.8.17"
```
Comment (jerboaa) [inline ubi8-openjdk-21.yaml:50]: `3.8.21`?

**Review by jerboaa — APPROVED** [review]

File: ubi8-openjdk-21.yaml (hunk 2 above)
Comment (jmtd) [reply ubi8-openjdk-21.yaml:50]: Good catch. At the moment there is no `maven-openjdk21` RPM package, which would be what `3.8.21` should install (once the module exists), but we expect there to be in the future. So I think for now it's ok to use the 17 one in conjunction with singleton-jdk.

File: ubi8-openjdk-21.yaml (hunk 1 above)
Comment (jmtd) [reply ubi8-openjdk-21.yaml]: Thank you yes: resolved in 4eff5d8d6293b84de7221c2b8551e122390fcf0b.

File: ubi8-openjdk-21.yaml (hunk 2 above)
Comment (jerboaa) [reply ubi8-openjdk-21.yaml:50]: OK.

Comment (sefroberg) [reply ubi8-openjdk-21.yaml:50]: @jmtd is there a ticket to track the creation of maven-openjdk21?

Comment (jmtd) [reply ubi8-openjdk-21.yaml:50]: There is for RHEL 9.4 (RHEL-13046), but not (yet) for 8.10. I asked the filer of the former whether they'd like me to clone for the latter and haven't yet had a response.

**Pull request merged**